gcloud/bigquery/job.py

```diff
@@ -0,0 +1,702 @@
+    :param destination: Table into which data is to be loaded.
+
+    :type source_uris: sequence of string
+    :param source_uris: URIs of data files to be loaded.
+
+    :type client: :class:`gcloud.bigquery.client.Client`
+    :param client: A client which holds credentials and project configuration
+                   for the dataset (which requires a project).
+
+    :type schema: list of :class:`gcloud.bigquery.table.SchemaField`
+    :param schema: The job's schema
+    """
+    def __init__(self, name, destination, source_uris, client, schema=()):
+        self.name = name
+        self.destination = destination
+        self._client = client
+        self.source_uris = source_uris
+        self.schema = schema
+        self._properties = {}
+        self._configuration = _LoadConfiguration()
+
+    @property
+    def project(self):
+        """Project bound to the job.
+
+        :rtype: string
+        :returns: the project (derived from the client).
+        """
+        return self._client.project
+
+    @property
+    def path(self):
+        """URL path for the job's APIs.
+
+        :rtype: string
+        :returns: the path based on project and job name.
+        """
+        return '/projects/%s/jobs/%s' % (self.project, self.name)
+
+    @property
+    def schema(self):
+        """Table's schema.
+
+        :rtype: list of :class:`SchemaField`
+        :returns: fields describing the schema
+        """
+        return list(self._schema)
+
+    @schema.setter
+    def schema(self, value):
+        """Update table's schema
+
+        :type value: list of :class:`SchemaField`
+        :param value: fields describing the schema
+
+        :raises: TypeError if 'value' is not a sequence, or ValueError if
+                 any item in the sequence is not a SchemaField
+        """
+        if not all(isinstance(field, SchemaField) for field in value):
+            raise ValueError('Schema items must be fields')
+        self._schema = tuple(value)
+
+    @property
+    def etag(self):
+        """ETag for the job resource.
+
+        :rtype: string, or ``NoneType``
+        :returns: the ETag (None until set from the server).
+        """
+        return self._properties.get('etag')
+
+    @property
+    def job_id(self):
+        """ID for the job resource.
+
+        :rtype: string, or ``NoneType``
+        :returns: the ID (None until set from the server).
+        """
+        return self._properties.get('id')
+
+    @property
+    def self_link(self):
+        """URL for the job resource.
+
+        :rtype: string, or ``NoneType``
+        :returns: the URL (None until set from the server).
+        """
+        return self._properties.get('selfLink')
+
+    @property
+    def user_email(self):
+        """E-mail address of user who submitted the job.
+
+        :rtype: string, or ``NoneType``
+        :returns: the URL (None until set from the server).
+        """
+        return self._properties.get('user_email')
```

Comment:
Lulz at Google

@jgeewax There should probably be an internal bug for `user_email` not being camel case like every other property name